microservices/matchmaking-service/src/scrim/scrim.event-subscriber.ts

```diff
@@ -1,33 +1,105 @@
+                        break;
+                    default:
+                        break;
+                }
+            });
         });
+
     }
 
-    onSubmissionStarted = async (d: EventResponse<EventTopic.SubmissionStarted>): Promise<void> => {
-        // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+    onSubmissionStarted = async (d: EventResponse<EventTopic.SubmissionStarted | EventTopic>): Promise<void> => {
         if (d.topic !== EventTopic.SubmissionStarted) return;
-        const targetSubmissionId = d.payload.submissionId;
+        const {payload} = d as EventResponse<EventTopic.SubmissionStarted>;
+        const targetSubmissionId = payload.submissionId;
         const scrim = await this.scrimCrudService.getScrimBySubmissionId(targetSubmissionId);
         if (!scrim) return;
         await this.scrimCrudService.updateScrimStatus(scrim.id, ScrimStatus.SUBMITTING);
         scrim.status = ScrimStatus.SUBMITTING;
         await this.eventsService.publish(EventTopic.ScrimUpdated, scrim, scrim.id);
     };
+
+    onSubmissionReset = async (d: EventResponse<EventTopic.SubmissionReset | EventTopic>): Promise<void> => {
+        if (d.topic !== EventTopic.SubmissionReset) return;
+        const {payload} = d as EventResponse<EventTopic.SubmissionReset>;
+        const targetSubmissionId = payload.submissionId;
+        const scrim = await this.scrimCrudService.getScrimBySubmissionId(targetSubmissionId);
+        if (!scrim) return;
+        await this.scrimCrudService.updateScrimStatus(scrim.id, ScrimStatus.IN_PROGRESS);
+        scrim.status = ScrimStatus.IN_PROGRESS;
+        await this.eventsService.publish(EventTopic.ScrimUpdated, scrim, scrim.id);
+    };
+
+    onSubmissionComplete = async (d: EventResponse<EventTopic.SubmissionComplete | EventTopic>): Promise<void> => {
+        if (d.topic !== EventTopic.SubmissionComplete) return;
+        const {payload} = d as EventResponse<EventTopic.SubmissionComplete>;
+
+        const scrim = await this.scrimCrudService.getScrimBySubmissionId(payload.submissionId);
+        if (!scrim) {
+            this.logger.warn(`Scrim not found for submission ${payload.submissionId}`);
+            return;
+        }
+        await this.scrimService.moveToRatification(scrim.id);
+    };
+
+    onSubmissionRatified = async (d: EventResponse<EventTopic.SubmissionRatified | EventTopic>): Promise<void> => {
+        if (d.topic !== EventTopic.SubmissionRatified) return;
+        const {payload} = d as EventResponse<EventTopic.SubmissionRatified>;
+
+        const scrim = await this.scrimCrudService.getScrimBySubmissionId(payload.submissionId);
+        if (!scrim) {
+            this.logger.warn(`Scrim not found for submission ${payload.submissionId}`);
+            return;
+        }
+        await this.scrimService.completeScrim(scrim.id);
+    };
+
+    onSubmissionRejected = async (d: EventResponse<EventTopic.SubmissionRejected | EventTopic>): Promise<void> => {
```

Comment:
A lot of repeated code here in a pattern. Seems odd that we'd switch on the event type, to call this boilerplate functions, that effectively just move a state machine forward once errors are handled. There's a cleaner way to implement state machines for sure, not necessary for this effort though.